src/argparcel/__init__.py

```diff
@@ -261,3 +261,18 @@ def parse[T: _typeshed.DataclassInstance](
     }
 
     return cls(**converted_kwargs)
+
+
+def uses_types[T](*types: type) -> Callable[[T], T]:
+    """Decorate a dataclass, and indicate types needed at runtime.
+
+    The existence of this method is a somewhat disgusting workaround to let you, the
+    user, declare to your static linting tools that a type is required at runtime, and
+    that the import must not be moved into a `TYPE_CHECKING` block.
+    """
+    del types
+
+    def f[_T](cls: _T) -> _T:
```

Comment:
ty moans about this

```
error: lint:invalid-return-type: Return type does not match returned value
   --> src/argparcel/__init__.py:278:12
    |
276 |         return cls
277 |
278 |     return f
    |            ^ Expected `(T, /) -> T`, found `def f(cls: _T) -> _T`
    |
   ::: src/argparcel/__init__.py:266:36
    |
266 | def uses_types[T](*types: type) -> Callable[[T], T]:
    |                                    ---------------- Expected `(T, /) -> T` because of return type
```

Reply:
It's not wrong!

Reply:
Does `def f(cls: T) -> T:` work?

Reply:
but I couldn't fix it 91751f693108493a68e40c014f31985f3b401794

Reply:
Aha, the `/` matters!

Reply:
Oh I'm being silly, yes I can: 14f14d8ff514377f5a55b8967ed3c4694621d578

Reply:
I think @tpgillam has not refreshed the page to see my replies ;)

Reply:
But well done. Looks good.

Reply:
@Rupt ha yes, github doesn't dynamically load new comments unless the page is forcefully refreshed. I fall foul of this often!

Reply:
> Aha, the / matters!

I thought it did - but it turns out it did not (I tried removing it again after). It seems to me a non-trivial decision must have been made to decide that the two should no be distinguished...